pallets/parachain-staking/src/lib.rs

```diff
@@ -1426,26 +1461,26 @@ pub mod pallet {
 
 			// prepare update of collator state
 			let mut state = <CollatorState<T>>::get(&collator).ok_or(Error::<T>::CandidateNotFound)?;
+			let num_delegations_pre_insertion: u32 = state.delegators.len().saturated_into();
+
 			ensure!(!state.is_leaving(), Error::<T>::CannotDelegateIfLeaving);
 			let delegation = Stake {
 				owner: acc.clone(),
 				amount,
 			};
+
+			// attempt to insert delegator and check for uniqueness, excess is handled below
+			let insert_delegator = state
```

Comment:
Note: As for the collators, this throws if insertion would exceed the bounded vec's max size. Returns true if the item is unique in the set, otherwise returns false.

In contrast to collators, we handle replacing delegators if the bound limit is met. Thus, we handle the error case below in `do_update_delegator`.